osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerSpectateButton.cs

```diff
@@ -111,6 +111,8 @@ private void updateState()
 
         private CancellationTokenSource? downloadCheckCancellation;
 
+        private int? lastDownloadCheckedBeatmapId;
+
         private void checkForAutomaticDownload()
         {
             downloadCheckCancellation?.Cancel();
```

Comment:
I'm pretty sure I made this exact change in this PR during my refactorings, but was rejected at the time. And in that PR, I'm positive I noted that this cancellation should no longer be done here, and instead be moved below the check you added.

Can you confirm the behaviour here?